activesupport/lib/active_support/testing/composite_filter.rb

```diff
@@ -18,14 +18,19 @@ def derive_regexp(filter)
         end
 
         def derive_line_filters(patterns)
-          patterns.map do |file_and_line|
-            file, line = file_and_line.split(':')
-            Filter.new(@runnable, file, line) if file
+          patterns.flat_map do |file_and_line|
+            file, *lines = file_and_line.split(':')
+
+            if lines.empty?
+              Filter.new(@runnable, file) if file
+            else
+              lines.map { |line| Filter.new(@runnable, file, line) }
+            end
           end
         end
 
         class Filter # :nodoc:
-          def initialize(runnable, file, line)
+          def initialize(runnable, file, line = nil)
```

Comment:
Why do we need to default the argument? Far as I can tell, nil can be passed through before.